consensus/polybft/state_sync_manager.go

```diff
@@ -362,14 +362,18 @@ func (s *stateSyncManager) PostEpoch(req *PostEpochRequest) error {
 }
 
 type PostBlockRequest struct {
-	// Block is a reference of the executed block
-	Block *types.Block
+	// FullBlock is a reference of the executed block
+	FullBlock *types.FullBlock
+	// Epoch is the epoch number of the executed block
+	Epoch uint64
+	// IsEpochEndingBlock indicates if this is the last block in epoch
+	IsEpochEndingBlock bool
```

Comment:
Note that per the RFC doc, there is already another handler for `PostEpoch`. Would that work for this use case?

Reply:
That is something I wanted to write in the PR description. I opted this for simplicity and to avoid two writes to `boltDb`. Basically if an exit event happens in the last block in epoch, we won't know if it was finalized, so to avoid errors, and possible malicious exits, we will include all exits that happened at the epoch ending blocks in the next epoch (they will be part of exit root of the first block in next epoch). Since `consensus_runtime` already knows on `OnBlockInserted` if the block is an epoch ending block, I just added this field, so that I can save the exit events right away in `boltDb` as part of the next epoch.

Alternative to this is to remove this field, and on `PostBlock` just save these events in the epoch they occurred. But, on the `PostEpoch`, `checkpointManager` would need to see if there are any exit events saved in the last block of previous epoch, and update them in `boltDb` to the new epoch. This is doable, but it means we would have additional update transactions to `boltDb`.

Reply:
I agree, it should be uniformed. I'll change this and implement the alternative approach I mentioned in previous comment, where we will handle this in `PostEpoch` implementation in `checkpointManager`.

Reply:
Let's do that and If you feel it gets tricky let's revisit the assumptions. I wrote them from a high-level perspective.

Reply:
Done: https://github.com/0xPolygon/polygon-edge/pull/1113/commits/817c6c54e162ac9dffe91e92805331333f69587d